modules/lightmapper_rd/lightmapper_rd.cpp

```diff
@@ -470,7 +471,23 @@ void LightmapperRD::_create_acceleration_structures(RenderingDevice *rd, Size2i
 			t.max_bounds[0] = taabb.position.x + MAX(taabb.size.x, 0.0001);
 			t.max_bounds[1] = taabb.position.y + MAX(taabb.size.y, 0.0001);
 			t.max_bounds[2] = taabb.position.z + MAX(taabb.size.z, 0.0001);
-			t.pad0 = t.pad1 = 0; //make valgrind not complain
+
+			t.cull_mode = RendererRD::MaterialStorage::ShaderData::CULL_BACK;
+
+			RID material = mi.data.material[i];
+			if (material.is_valid()) {
+				// TODO: Switch based on rendering method.
+				if (OS::get_singleton()->get_current_rendering_method() == "gl_compatibility") {
+					GLES3::MaterialStorage *material_storage = GLES3::MaterialStorage::get_singleton();
```

Comment:
We shouldn't be accessing the internals of the renderer like this. We need to expose it though the RenderingServer. See `shader_get_code()` for an example of how that is done. 

Even better, we can avoid moving CullMode into ShaderData and just check the shader's mode, i.e.:
```
String MaterialStorage::shader_get_cull_mode(RID p_shader) const {
	const GLES3::Shader *shader = shader_owner.get_or_null(p_shader);
	ERR_FAIL_NULL_V(shader, RS::CULL_DISABLED);
	if (shader->mode == RS::SHADER_SPATIAL && shader->data) {
		return shader->data->cull_mode; // Need to do a cast here to ensure it is SceneShaderData
	}
	return RS::CULL_DISABLED; // RS::CullMode needs to be added
}
```
Then you can also remove the `Cull` enum from `SceneShaderData` and use the `RS::CullMode` directly

Reply:
Fixed. Although `shader_get_cull_mode` in MaterialStorage (RD) has to do two casts, not sure if there's a better way: 
```cpp
RS::CullMode MaterialStorage::shader_get_cull_mode(RID p_shader) const {
	const RendererRD::MaterialStorage::Shader *shader = shader_owner.get_or_null(p_shader);
	ERR_FAIL_NULL_V(shader, RS::CULL_MODE_DISABLED);
	if (shader->type == ShaderType::SHADER_TYPE_3D && shader->data) {
		RendererSceneRenderImplementation::SceneShaderForwardClustered::ShaderData *sd_clustered = dynamic_cast<RendererSceneRenderImplementation::SceneShaderForwardClustered::ShaderData *>(shader->data);
		if (sd_clustered) {
			return (RS::CullMode)sd_clustered->cull_mode;
		}

		RendererSceneRenderImplementation::SceneShaderForwardMobile::ShaderData *sd_mobile = dynamic_cast<RendererSceneRenderImplementation::SceneShaderForwardMobile::ShaderData *>(shader->data);
		if (sd_mobile) {
			return (RS::CullMode)sd_mobile->cull_mode;
		}
	}
	return RS::CULL_MODE_DISABLED;
}
```